src/compiler-debugging.md

```diff
@@ -150,65 +150,94 @@ Cool, now I have a backtrace for the error!
+particular module.
+
+For example, to get the `debug!` output for a specific module, you can run the
+compiler with `RUSTC_LOG=path::to::module=debug rustc my-file.rs`. All `debug!`
+output will then appear in standard error.
 
-If you are developing rustdoc, use `RUSTDOC_LOG` instead.
+If you are developing rustdoc, use `RUSTDOC_LOG` instead. If you are developing
+Miri, use `MIRI_LOG` instead. You get the idea :)
 
-See the [env-logger] doc for more info on the full syntax. (Note: unlike the
-compiler, the env-logger crate and its examples use the `RUST_LOG` env
-variable.)
+See the [`tracing`] crate's docs, and specifically the docs for [`debug!`] to
+see the full syntax you can use. See the [env-logger] doc for more info on the
+full syntax. (Note: unlike the compiler, the [`tracing`] crate and its examples
+use the `RUST_LOG` environment variable. rustc, rustdoc, and other tools set
+custom environment variables.)
 
 **Note that unless you use a very strict filter, the logger will emit a lot of
 output, so use the most specific module(s) you can (comma-separated if
 multiple)**. It's typically a good idea to pipe standard error to a file and
 look at the log output with a text editor.
 
-So to put it together.
+So, to put it together:
 
 ```bash
 # This puts the output of all debug calls in `rustc_middle/src/traits` into
 # standard error, which might fill your console backscroll.
-$ RUSTC_LOG=rustc_middle::traits rustc +stage1 my-file.rs
+$ RUSTC_LOG=rustc_middle::traits=debug rustc +stage1 my-file.rs
 
 # This puts the output of all debug calls in `rustc_middle/src/traits` in
 # `traits-log`, so you can then see it with a text editor.
-$ RUSTC_LOG=rustc_middle::traits rustc +stage1 my-file.rs 2>traits-log
+$ RUSTC_LOG=rustc_middle::traits=debug rustc +stage1 my-file.rs 2>traits-log
 
-# Not recommended. This will show the output of all `debug!` calls
+# Not recommended! This will show the output of all `debug!` calls
 # in the Rust compiler, and there are a *lot* of them, so it will be
 # hard to find anything.
 $ RUSTC_LOG=debug rustc +stage1 my-file.rs 2>all-log
 
-# This will show the output of all `info!` calls in `rustc_trans`.
+# This will show the output of all `info!` calls in `rustc_codegen_ssa`.
 #
-# There's an `info!` statement in `trans_instance` that outputs
+# There's an `info!` statement in `codegen_instance` that outputs
 # every function that is translated. This is useful to find out
 # which function triggers an LLVM assertion, and this is an `info!`
 # log rather than a `debug!` log so it will work on the official
 # compilers.
-$ RUSTC_LOG=rustc_trans=info rustc +stage1 my-file.rs
+$ RUSTC_LOG=rustc_codegen_ssa=info rustc +stage1 my-file.rs
 
-# This will show the output of all `info!` calls made by rustdoc or any rustc library it calls.
+# This will show the output of all `info!` calls made by rustdoc
+# or any rustc library it calls.
 $ RUSTDOC_LOG=info rustdoc +stage1 my-file.rs
```

Comment:
This has a lot of overwhelming output because of all the rustc_metadata logging:
```
$ RUSTDOC_LOG=info rustdoc ~/test-rustdoc/iter.rs 2>&1  | wc -l
8365
```

The one I do actually use (for the same reason as rustc_codegen_ssa) is `RUSTDOC_LOG=rustdoc::passes::collect_intra_doc_links=info`:

```
$ RUSTDOC_LOG=rustdoc::passes::collect_intra_doc_links=info rustdoc +beta ~/test-rustdoc/iter.rs
 INFO rustdoc::passes::collect_intra_doc_links ignoring warning from parent crate: unresolved link to `Self::st_atime`
 INFO rustdoc::passes::collect_intra_doc_links ignoring warning from parent crate: unresolved link to `Self::st_mtime`
 INFO rustdoc::passes::collect_intra_doc_links ignoring warning from parent crate: unresolved link to `Self::st_ctime`
```
You might also mention that you can shorten that to `rustdoc::passes::collect` and it will still work.

Reply:
Oh, I just saw that you compare this to `RUSTDOC_LOG=rustdoc` a line later. Maybe this isn't necessary then.

Reply:
> You might also mention that you can shorten that to rustdoc::passes::collect and it will still work.

I'll mention that.